pywr/nodes.py

```diff
@@ -862,3 +862,47 @@ def after(self, timestep):
         self.commit_all(self.link.flow)
 
 from pywr.domains.river import *
+
+
+class DelayNode(Node):
+    """ A node that delays flow for a given number of timesteps or days
+
+    Parameters
+    ----------
+    model : `pywr.model.Model`
+    name : string
+        Name of the node.
+    timesteps: int
+        Number of timesteps to delay the flow.
+    days: int
+        Number of days to delay the flow. Specifying a number of days (instead of a number
+        of timesteps) is only valid if the number of days is exactly divisible by the model timestep delta.
+    """
+
+    def __init__(self, model, name, **kwargs): 
+        self.allow_isolated = True
+        output_name = "{} - output".format(name)
```

Comment:
Can we follow the naming convention in `PiecewiseLink` for these internal nodes?